app/services/support_interface/applications_by_demographic_domicile_and_degree_class_export.rb

```diff
@@ -0,0 +1,196 @@
+                    FROM application_forms
+                    AS subsequent_application_forms
+                    WHERE f.id = subsequent_application_forms.previous_application_form_id
+                  )
+                )
+            GROUP BY
+                c.id, f.id, age_group, degree_class, domicile, sex
+        )
+        SELECT
+            raw_data.age_group[2],
+            raw_data.sex,
+            raw_data.ethnicity,
+            raw_data.disability,
+            raw_data.degree_class[2],
+            raw_data.domicile,
+            COUNT(*),
+            raw_data.status[2]
+        FROM
+            raw_data
+        GROUP BY
+            raw_data.age_group,
+            raw_data.sex,
+            raw_data.ethnicity,
+            raw_data.disability,
+            raw_data.degree_class,
+            raw_data.domicile,
+            raw_data.status
+        ORDER BY
+            raw_data.age_group[1],
+            raw_data.degree_class[1]
+      SQL
+    end
+
+    def age_group_sql
+      "WHEN f.date_of_birth > '#{Date.new(RecruitmentCycle.current_year - 25, 8, 31)}' THEN ARRAY['0', 'Under 25']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 30, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 25, 8, 31)}' THEN ARRAY['1', '25 to 29']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 35, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 30, 8, 31)}' THEN ARRAY['2', '30 to 34']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 40, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 35, 8, 31)}' THEN ARRAY['3', '35 to 39']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 45, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 40, 8, 31)}' THEN ARRAY['4', '40 to 44']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 50, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 45, 8, 31)}' THEN ARRAY['5', '45 to 49']
+      WHEN f.date_of_birth BETWEEN '#{Date.new(RecruitmentCycle.current_year - 55, 9, 1)}' AND '#{Date.new(RecruitmentCycle.current_year - 50, 8, 31)}' THEN ARRAY['6', '50 to 54']
+      WHEN f.date_of_birth < '#{Date.new(RecruitmentCycle.current_year - 55, 9, 1)}' THEN ARRAY['7', '55 and over']"
+    end
+
+    def equality_and_diversity_sql
+      "WHEN f.equality_and_diversity is NULL THEN 'Not provided'
+      WHEN f.equality_and_diversity->> 'sex' = 'intersex' then 'Intersex'
+      WHEN f.equality_and_diversity->> 'sex' = 'male' then 'Male'
+      WHEN f.equality_and_diversity->> 'sex' = 'female' then 'Female'
+      WHEN f.equality_and_diversity->> 'sex' = 'Prefer not to say' then 'Prefer not to say'"
+    end
+
+    def domicile_sql
+      "WHEN f.country = 'GB' THEN 'UK'
+      WHEN f.country IN (#{EU_COUNTRY_CODES.map { |country| "'#{country}'" }.join(',')}) THEN 'EU'
+      ELSE
+     'Non EU'"
+    end
+
+    def degree_class_sql
```

Comment:
I think we can probably use the aggr syntax here to deal with retuning the highest grade here 

 `"WHEN 'First class honours'  = ANY(ARRAY_AGG(q.grade)) THEN ARRAY['0', 'First class honours']`

Reply:
Managed to resolve the `MIN(raw_data.degree_class[1]) AS degree_class` on line 143. couldn't figure out how to get it to work without assigning the values to an array with multiple elements. 🤔